cargo-marker/src/backend/toolchain.rs

```diff
@@ -53,108 +55,114 @@ impl Toolchain {
+
+        let _span = info_span!("search_next_to_cargo_marker", path = %driver_path.display()).entered();
+
+        info!("Searching for driver");
+
+        if !driver_path.is_file() {
+            return Err(Error::root(format!(
+                "Could not find driver next to the cargo-marker binary at {}",
+                driver_path.display().red().bold()
+            )));
         }
 
-        Err(ExitStatus::MissingDriver)
+        info!("Found driver");
+
+        Ok(Toolchain {
+            driver_path,
+            cargo: Cargo::default(),
+        })
     }
 }
 
-pub(crate) fn get_toolchain_folder(toolchain: &str) -> Result<PathBuf, ExitStatus> {
-    if let Ok(toolchain_cargo) = rustup_which(toolchain, "cargo", false) {
-        // ../toolchain/bin/cargo -> ../toolchain
-        if let Some(path) = toolchain_cargo.ancestors().nth(2) {
-            return Ok(path.to_path_buf());
-        }
-    }
-    Err(ExitStatus::BadConfiguration)
+pub(crate) fn get_toolchain_folder(toolchain: &str) -> Result<PathBuf> {
+    let toolchain_cargo = rustup_which(toolchain, "cargo")?;
+
+    // ../toolchain/bin/cargo -> ../toolchain
+    let path = toolchain_cargo.ancestors().nth(2).context(|| {
+        format!(
+            "Unexpected layout of the rustup toolchain binary dir. There are not \
+            enough ancestors in the path `{}`",
+            toolchain_cargo.display()
+        )
+    })?;
+
+    Ok(path.to_path_buf())
 }
 
-pub(crate) fn rustup_which(toolchain: &str, tool: &str, verbose: bool) -> Result<PathBuf, ExitStatus> {
-    if verbose {
-        println!("Searching for `{tool}` with rustup for toolchain `{toolchain}`");
-    }
+pub(crate) fn rustup_which(toolchain: &str, tool: &str) -> Result<PathBuf> {
+    let mut cmd = Command::new("rustup");
+    cmd.args(["which", "--toolchain", toolchain, tool]);
 
-    let output = Command::new("rustup")
-        .args(["which", "--toolchain", toolchain, tool])
-        .output()
-        .map_err(|err| ExitStatus::fatal(err, "failed to execute rustup"))?;
+    let output = cmd.log().output().context(|| "failed to execute rustup")?;
 
     if !output.status.success() {
         let stderr = String::from_utf8_lossy(&output.stderr);
-        return Err(ExitStatus::fatal(
-            stderr.trim(),
-            format!("failed to execute `rustup which` for `{tool}` with `{toolchain}` toolchain"),
-        ));
+
+        return Err(Error::wrap(stderr.trim(), format!("Command failed: {}", cmd.display())));
     }
 
-    let string_path = String::from_utf8(output.stdout).map_err(|err| ExitStatus::fatal(err, "incorrect bytes"))?;
-    let path = PathBuf::from_str(string_path.trim())
-        .map_err(|err| ExitStatus::fatal(err, format!("failed to parse path for `{tool}`")))?;
+    let string_path = String::from_utf8(output.stdout).context(|| "incorrect bytes")?;
+    let path = PathBuf::from_str(string_path.trim()).context(|| format!("failed to parse path for `{tool}`"))?;
```

Comment:
Ah, in this case it's possible to use `PathBuf::from` instead of `from_str`, because the operation of converting from string to `PathBuf` is infallible.

Anyway, you are right, fixed the capitalization